jujusvg.go

```diff
@@ -45,28 +46,52 @@ func NewFromBundle(b *charm.BundleData, iconURL func(*charm.Reference) string, f
 	}
 	sort.Strings(serviceNames)
 	services := make(map[string]*service)
+	servicesNeedingPlacement := make(map[string]bool)
 	for _, name := range serviceNames {
+		needsPlacement := false
 		serviceData := b.Services[name]
 		x, xerr := strconv.ParseFloat(serviceData.Annotations["gui-x"], 64)
 		y, yerr := strconv.ParseFloat(serviceData.Annotations["gui-y"], 64)
 		if xerr != nil || yerr != nil {
-			return nil, errgo.Newf("service %q does not have a valid position", name)
+			if serviceData.Annotations["gui-x"] == "" && serviceData.Annotations["gui-y"] == "" {
```

Comment:
I thought about it, but an x without a y still falls under the invalid clause below.